### PR TITLE
Improved handling of invalid key strings

### PR DESCRIPTION
**Pull request by macfarla**
BLSPubKey.fromHexString now forces early validation of the key string. This means if bad keys are passed to the API they are validated there and not passed to the back end.

### CODE REVIEW

**Review by rolfyone — APPROVED** [review]
LGTM

File: data/provider/src/main/java/tech/pegasys/artemis/api/schema/BLSPubKey.java [hunk 1]
```diff
@@ -60,8 +61,11 @@ public String toString() {
     return bytes.toString();
   }
 
-  public static BLSPubKey fromHexString(String value) {
-    return new BLSPubKey(Bytes.fromHexString(value));
+  public static BLSPubKey fromHexString(String value) throws IllegalArgumentException {
+    BLSPublicKey blsPublicKey = BLSPublicKey.fromBytes(Bytes.fromHexString(value));
+    // force validation early - this will throw IllegalArgumentException if not valid
+    blsPublicKey.getPublicKey().g1Point();
```
Comment (ajsutton) [inline data/provider/src/main/java/tech/pegasys/artemis/api/schema/BLSPubKey.java]: nit: Would be a lot better if this were wrapped in a `blsPublicKey.isValid()`

File: data/provider/src/main/java/tech/pegasys/artemis/api/schema/BLSPubKey.java [hunk 2]
```diff
@@ -60,8 +61,11 @@ public String toString() {
     return bytes.toString();
   }
 
-  public static BLSPubKey fromHexString(String value) {
-    return new BLSPubKey(Bytes.fromHexString(value));
+  public static BLSPubKey fromHexString(String value) throws IllegalArgumentException {
```
Comment (ajsutton) [inline data/provider/src/main/java/tech/pegasys/artemis/api/schema/BLSPubKey.java]: nit: Don't need to declare unchecked exceptions like `IllegalArgumentException`.

**Review by ajsutton — APPROVED** [review]
LGTM. #1461 will also need to check the validity of BLS keys so would be good to have it provide an `isValid` method rather than depending on the internal implementation detail that `g1Point` happens to cause it to parse.

**Pull request merged**